azurerm/resource_arm_application_gateway.go

```diff
@@ -1560,16 +1636,86 @@ func flattenApplicationGatewayConnectionDraining(input *network.ApplicationGatew
 }
 
 func expandApplicationGatewaySslPolicy(d *schema.ResourceData) *network.ApplicationGatewaySslPolicy {
-	vs := d.Get("disabled_ssl_protocols").([]interface{})
-	results := make([]network.ApplicationGatewaySslProtocol, 0)
+	policy := network.ApplicationGatewaySslPolicy{}
+	vs := d.Get("ssl_policy").([]interface{})
+	if len(vs) == 0 {
+		return &policy
+	}
+	v := vs[0].(map[string]interface{})
+
+	disabledSSLPolicies := make([]network.ApplicationGatewaySslProtocol, 0)
+	for _, policy := range v["disabled_protocols"].([]interface{}) {
+		disabledSSLPolicies = append(disabledSSLPolicies, network.ApplicationGatewaySslProtocol(policy.(string)))
+	}
+	if len(disabledSSLPolicies) == 0 {
+		for _, policy := range d.Get("disabled_ssl_protocols").([]interface{}) {
+			disabledSSLPolicies = append(disabledSSLPolicies, network.ApplicationGatewaySslProtocol(policy.(string)))
+		}
+	}
+
+	if len(disabledSSLPolicies) > 0 {
+		policy = network.ApplicationGatewaySslPolicy{
+			DisabledSslProtocols: &disabledSSLPolicies,
+		}
```

Comment:
due to the way Azure's API works (in that it fields with nil values are ignored, rather than reset), this needs to be set in either case (even if it's an empty array)